src/dune_rules/compilation_context.ml

```diff
@@ -185,9 +204,14 @@ let for_alias_module t =
   ; includes = Includes.empty
   ; stdlib = None
   ; sandbox
+  ; dep_graphs =
+      (if Modules.is_stdlib_alias t.modules alias_module then
+        t.dep_graphs
```

Comment:
This looks a bit shady and I couldn't easily convince myself that ignoring `t.dep_graph` in the else branch is the right thing to do.

I guess there is some kind of invariant between `t.modules` and `t.dep_graph`, which makes `t.dep_graph` irrelevant if a certain condition holds about `t.modules`. This makes me think that `dep_graph` should instead live inside `t.modules` but I'm not sure if this is going to create problems.

Reply:
I'll add a comment explaining why we do this.

Reply:
I ended up enforcing the invariant that `dep_graphs` correspond to the dep graphs of the `modules` field. I packed them together in a separate record so that we remember to update them together. Agreed that `dep_graphs` could move to `Modules.t` directly, I'll leave that as a future improvement for now.

Please have a look at the last commit.

Reply:
Nice, that looks cleaner!